build.sbt

```diff
@@ -1,250 +1,9 @@
-  .settings(commonSettings)
-  .dependsOn(utils, dynamoFormatters)
-  .settings(
-    libraryDependencies ++= Seq(
-      dynamoClient,
-      snsClient
-    )
-  )
-
-lazy val ingestParentFolderOpexCreator = (project in file("ingest-parent-folder-opex-creator"))
-  .settings(commonSettings)
-  .dependsOn(utils)
-  .settings(
-    libraryDependencies ++= Seq(
-      s3Client,
-      fs2Core,
-      reactorTest % Test,
-      scalaXml
-    )
-  )
-
-lazy val ingestUpsertArchiveFolders = (project in file("ingest-upsert-archive-folders"))
-  .settings(commonSettings)
-  .dependsOn(utils, dynamoFormatters)
-  .settings(
-    libraryDependencies ++= Seq(
-      dynamoClient,
-      eventBridgeClient,
-      preservicaClient,
-      pureConfig
-    )
-  )
-
-lazy val ingestWorkflowMonitor = (project in file("ingest-workflow-monitor"))
-  .settings(commonSettings)
-  .dependsOn(utils)
-  .settings(
-    libraryDependencies += preservicaClient
-  )
-
-lazy val preservicaConfig = (project in file("preservica-config"))
-  .settings(commonSettings)
-  .dependsOn(utils)
-  .settings(
-    libraryDependencies ++= Seq(
-      preservicaClient,
-      s3Client,
-      scalaXml,
-      scalaParserCombinators,
-      jaxb
-    )
-  )
-
-lazy val ingestFolderOpexCreator = (project in file("ingest-folder-opex-creator"))
-  .settings(commonSettings)
-  .dependsOn(utils, dynamoFormatters)
-  .settings(
-    libraryDependencies ++= Seq(
-      awsCrt,
-      fs2Core,
-      preservicaClient,
-      scalaXml,
-      dynamoClient,
-      s3Client
-    )
-  )
-
-lazy val startWorkflow = (project in file("ingest-start-workflow"))
-  .settings(commonSettings)
-  .dependsOn(utils)
-  .settings(
-    libraryDependencies += preservicaClient
-  )
-
-lazy val entityEventGenerator = (project in file("entity-event-generator-lambda"))
-  .settings(commonSettings)
-  .dependsOn(utils)
-  .settings(
-    libraryDependencies ++= Seq(
-      awsSecretsManager,
-      catsEffect,
-      dynamoClient,
-      preservicaClient,
-      snsClient,
-      sttpClient,
-      typeSafeConfig
-    )
-  )
-
-lazy val getLatestPreservicaVersion = (project in file("get-latest-preservica-version-lambda"))
-  .settings(commonSettings)
-  .dependsOn(utils, dynamoFormatters)
-  .settings(
-    libraryDependencies ++= Seq(
-      dynamoClient,
-      snsClient,
-      preservicaClient
-    )
-  )
-
-lazy val ingestFindExistingAsset = (project in file("ingest-find-existing-asset"))
-  .settings(commonSettings)
-  .dependsOn(utils, dynamoFormatters)
-  .settings(
-    libraryDependencies ++= Seq(
-      dynamoClient,
-      preservicaClient
-    )
-  )
-
-lazy val rotatePreservationSystemPassword = (project in file("rotate-preservation-system-password"))
-  .settings(commonSettings)
-  .dependsOn(utils)
+  .aggregate(ingestLambdasRoot)
   .settings(
-    libraryDependencies ++= Seq(
-      preservicaClient,
-      secretsManagerClient
-    )
+    name := "preservation",
+    scalaVersion := "3.4.2"
```

Comment:
1. There is a Scala version in the `build.sbt` file; do we need to have them in both places?
2. How come the `name` here is "preservation" and in the `build.sbt` it's "lambdas"?

Reply:
1. No, we don't need them in both places, thanks. It just needs to be declared once in the root build.sbt
2. The file project with the name "preservation" will be an aggregate of the lambdas subproject and eventually the docker images (custodial copy) stuff as well. So it will be something like
```
.
├── build.sbt
├── python
│   └── lambdas
│       └── copy-files-from-tdr
└── scala
    ├── images
    │   └── build.sbt
    └── lambdas
        ├── build.sbt
        ├── dynamo-formatters
        ├── entity-event-generator-lambda
        ├── get-latest-preservica-version-lambda
        ├── ingest-asset-opex-creator
        ├── ingest-asset-reconciler
        ├── ingest-files-change-handler
        ├── ingest-find-existing-asset
         ....
```
So the name for the `images/build.sbt` will be dockerImages or something. I want to group things by how they're deployed. So the name in the root file needs to encompass everything we've got so I went with preservation.

Reply:
Ah, I see. Would "ingest" work or are we expecting to add non-ingest stuff here?

Reply:
No, we're not, ingest makes more sense. I'll change it.